### PR TITLE
Added CipherInterface and EncryptionTrait

### PR DESCRIPTION
**Pull request by jakeasmith**

### CODE REVIEW

File: lib/CipherInterface.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+interface CipherInterface
+{
+	/**
+	 * Encrypt the given text.
+	 *
+	 * @param string $plain_text The text to encrypt.
+	 *
+	 * @return string The encrypted text.
+	 */
+	function encrypt($plain_text);
```
Comment (jeremyjannotta) [inline lib/CipherInterface.php]: Was wondering if it made sense to support encryption/decryption of non-strings as well. For instance could encrypt entire arrays of data, or other types, but not sure how far down that hole to go.

**Comment by jeremyjannotta** [comment]
Interface looks good! :+1:

File: lib/CipherInterface.php (hunk 1 above)
Comment (jakeasmith) [inline lib/CipherInterface.php]: I don't think that's something we would want to support at this level. In the simple use case you can just `defineEncrypted(...)`. In a more complex use case where the value need to be stringified first you can handle that in the transformer itself and you have direct access to the `encrypt($plain_text)` and `decrypt($encrypted_text)` methods as you need them.

Comment (jakeasmith) [inline lib/CipherInterface.php]: As I was writing the above I was thinking about objects, but it probably does make sense to add a `defineEncryptedArray(...)` as well :)

**Comment by jakeasmith** [comment]
Thanks! I'll add tests

**Pull request closed without merging**